### PR TITLE
Task-1196: Added Travis and codecov coverage

### PR DESCRIPTION
**Pull request by arfathpasha**
@sychan Please integrate with kbase codecov account. Here are a list of changes you may want to make.

- change notifications recipient email
- change secure tokens (put two long standing tokens for CI). The tokens should be formatted as
  - TEST_TOKEN="first token"
  - TEST_TOKEN2="second token"
- remove branches develop and travis (these were used for testing). 
- change the codecov token (in "after-success" section) to point to the kbase codecov account

### CODE REVIEW

**Comment by MrCreosote** [comment]
Obviously this will always fail on PRs until auth can be run locally

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+- cd KBaseSearchEngine
+
+# create test_local dir
+- kb-sdk test || true
+
+# update test.cfg with travis user
+- sed -i "s/test_token=/test_token=$TEST_TOKEN\ntest.token2=$TEST_TOKEN2\n/" test_local/test.cfg
+- sed -i 's\https://appdev.kbase.us/services\https://ci.kbase.us/services\' test_local/test.cfg
+- sed -i 's\https://appdev.kbase.us/services/auth/api/legacy/KBase/Sessions/Login\https://ci.kbase.us/services/auth/api/legacy/KBase/Sessions/Login\'
+  test_local/test.cfg
```
[the remark this answers is not in the document]
Comment (arfathpasha) [reply .travis.yml]: These sed lines add the test tokens to test.cfg and change the urls  in test.cfg from the default appdev urls to ci urls

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+- pwd=$(pwd)
+- echo "#!/bin/bash" >> test_local/run_subjob.sh
+- echo -e "$pwd/test_local/run_docker.sh run --rm --user $(id -u) -v $pwd/test_local/subjobs/\$1/workdir:/kb/module/work
+  -v $pwd/test_local/workdir/tmp:/kb/module/work/tmp \$4 -e \"SDK_CALLBACK_URL=\$3\"
+  \$2 async" >> test_local/run_subjob.sh
+- cat test_local/run_subjob.sh
```
[the remark this answers is not in the document]
Comment (arfathpasha) [reply .travis.yml]: Removed.

File: test/src/kbasesearchengine/test/events/exceptions/RetrierTest.java [hunk 3]
```diff
@@ -168,8 +168,8 @@ public void consumer1RetrySuccessNoEvent() throws Exception {
         assertThat("incorrect retry count", le.retryCount, is(1));
         assertThat("incorrect event", le.event, is(Optional.absent()));
         TestCommon.assertExceptionCorrect(le.exception, new RetriableIndexingException("bar"));
-        assertCloseMS(start, le.time, 0, 20);
-        assertCloseMS(start, end, 50, 20);
+        assertCloseMS(start, le.time, 0, 30);
+        assertCloseMS(start, end, 50, 30);
```
[the remark this answers is not in the document]
Comment (arfathpasha) [reply test/src/kbasesearchengine/test/events/exceptions/RetrierTest.java]: Travis fails with 20

File: .travis.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (MrCreosote) [reply .travis.yml]: I'm specifically talking about the line 
`test_local/test.cfg`

Looking closer, I'm assuming it's supposed to be appended to the end of the previous line? Git is displaying it as being on a separate line.

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    recipients:
+    - arfathpasha@gmail.com
+    on_success: never
+    on_failure: always
+
+env:
+  global:
+  - MONGODB_VER=mongodb-linux-x86_64-2.6.12 ANT_TEST=test WIRED_TIGER=false
+  - JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
+  - secure: tguBGgSgTvVY/ayL0IFA08JM8Nu0ys0iwU2nte2oZIl8J6db+8FwzV6SlXeRjYYJYYHu3uOobYb2i1OpEGfRH2aE8+Q0TZzuupSlKb2cEReV9SAL4w0cb0azg06gjKmY+NoA18lCDZKV7X15Zh9cGkRvcM0Mjbhbbdek3n8vf1h5F1Gp6sV7MHSwQEGRmvmyEy/qTb7D1qIVEyZaxWV+ou3kd4eagiaJr0P+rvqBUOvoqWsUrroR3Vfs2A5J5gy4IeZJhpxwJsJz3UapuWcmYFV+fU/DgGdx3oaG7O5+rBH/v4uzZGaha7xuLFP93UURpxPbQyXKhsE1OiRW+9svpZw3yAS5/PN49GIWcqg3vNECkuyPBO9iSDLzz6iUNiErlsePFKKIlk/qzwOJimmz1+8ZCeDHZXUv81JK+RhKK1dlWECZcpYrr4YsTV4GVkVORIcseIy2jLmvRIer0pLiauOPooiNVcCv+TiYkwfeZdO8QiwsxN1WBzjzniwS3cA6z/AKmm0qJNpu37JTO19AiKEqgcjEfbftSp7S/nnIdGxObFt84iToGb3XfdFVeI8O3B9ngFxuxILb0hYfaXm5TfD2WIUL8pL6g3GVVapFJDkal1PTmc6uq6jU8UW+3SirqRd34ELHcoOKI03cnVdaAgCbKViTdGZtjZjpuBkHz48=
```
[the remark this answers is not in the document]
Comment (arfathpasha) [reply .travis.yml]: Done.

File: test/src/kbasesearchengine/test/events/exceptions/RetrierTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (MrCreosote) [reply test/src/kbasesearchengine/test/events/exceptions/RetrierTest.java]: OK, then we should change all the slop params in that test file from 20 -> 30

File: .travis.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (arfathpasha) [reply .travis.yml]: Yes, it's part of the previous line.

File: test/src/kbasesearchengine/test/events/exceptions/RetrierTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (arfathpasha) [reply test/src/kbasesearchengine/test/events/exceptions/RetrierTest.java]: Done.

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+# create test_local dir
+- kb-sdk test || true
+
+# update test.cfg with travis user
+- sed -i "s/test_token=/test_token=$TEST_TOKEN\ntest.token2=$TEST_TOKEN2\n/" test_local/test.cfg
+- sed -i 's\https://appdev.kbase.us/services\https://ci.kbase.us/services\' test_local/test.cfg
+- sed -i 's\https://appdev.kbase.us/services/auth/api/legacy/KBase/Sessions/Login\https://ci.kbase.us/services/auth/api/legacy/KBase/Sessions/Login\'
+  test_local/test.cfg
+- pwd=$(pwd)
```
Comment (MrCreosote) [inline .travis.yml]: pwd looks unused?

Comment (arfathpasha) [reply .travis.yml]: Removed

**Comment by MrCreosote** [comment]
Looks good to me. I think the ball is in @sychan 's court at this point.

**Comment by sychan** [comment]
@arfathpasha I don't have any codecov auth tokens, it looks like you put one into the auth2 repo. Are you using your token in the clear in there? Any reason why we can't just use travis to encrypt your existing codecov token into this repo as well?
In the meantime, I'll look at getting a codecov account

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    - arfathpasha@gmail.com
+    on_success: never
+    on_failure: always
+
+env:
+  global:
+  - MONGODB_VER=mongodb-linux-x86_64-2.6.12 ANT_TEST=test WIRED_TIGER=false
+  - JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
+  # TEST_TOKEN
+  - secure: tguBGgSgTvVY/ayL0IFA08JM8Nu0ys0iwU2nte2oZIl8J6db+8FwzV6SlXeRjYYJYYHu3uOobYb2i1OpEGfRH2aE8+Q0TZzuupSlKb2cEReV9SAL4w0cb0azg06gjKmY+NoA18lCDZKV7X15Zh9cGkRvcM0Mjbhbbdek3n8vf1h5F1Gp6sV7MHSwQEGRmvmyEy/qTb7D1qIVEyZaxWV+ou3kd4eagiaJr0P+rvqBUOvoqWsUrroR3Vfs2A5J5gy4IeZJhpxwJsJz3UapuWcmYFV+fU/DgGdx3oaG7O5+rBH/v4uzZGaha7xuLFP93UURpxPbQyXKhsE1OiRW+9svpZw3yAS5/PN49GIWcqg3vNECkuyPBO9iSDLzz6iUNiErlsePFKKIlk/qzwOJimmz1+8ZCeDHZXUv81JK+RhKK1dlWECZcpYrr4YsTV4GVkVORIcseIy2jLmvRIer0pLiauOPooiNVcCv+TiYkwfeZdO8QiwsxN1WBzjzniwS3cA6z/AKmm0qJNpu37JTO19AiKEqgcjEfbftSp7S/nnIdGxObFt84iToGb3XfdFVeI8O3B9ngFxuxILb0hYfaXm5TfD2WIUL8pL6g3GVVapFJDkal1PTmc6uq6jU8UW+3SirqRd34ELHcoOKI03cnVdaAgCbKViTdGZtjZjpuBkHz48=
```
Comment (sychan) [inline .travis.yml]: Please change this to:
  secure: "DC0WOpe4kkbV5QiMkhJLXpR+e1aCFHP0MPI7+adBSsEmaoIZPeN19jykD9DvATxTmlPkyRUilsMfJ+gx+P1Y4pJ0dfWsDXAgUOrf3g7QbPixEfrLBLqmY7oALcP6bKAIlZjgAtG4uFEmYJnZ9qP/5IWnzZsshJzscxxb7LWJok9hAaTs0rD5I6sHfru+/2mJnhcSE6WUhYbL88pm/3MJ0zIVoKMGPukYJY6YH6MzrMe///N5NScF6eqONFA0KVYp7PIHRAcWl82ivWhrN5Hrgq4p05+y8aq8eO8SNm4rbjayNX6Ya4zK8a5phv8zmIv0YCxlcB/e/vcLLMoXzoxZry7a7FwxCO/ao9v4kijNmOfrvp2fqRUl0OY7jQY2kMF7jowMBNoMzqweuTLohmTg2QCr/uaq6h3FxNwxesBRmHFnS4Bs3j6aQiomAuX5n+JunFG576wuFHih09dtu9ScUNr7rCOE5WCYQlsOuYFeigyv7/4iTZDr31OwwqnK48Bm+Nyw1vSp+274CjHlyCaMTSfd17kMQMVrfeRnzPSPEBQwXb98qzYxKeDEFXMKZZ5Cl+8BgxtQw5MDJ5ORiidJmfa7mewV5fK9MrIGRRZl+GTgE9MTJGpq0rEaAHtA309BbSNSXGN6VWfhj/E/sHNaqGzgRqn3PDnwu/4kPl8QVpc="

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  # TEST_TOKEN2
+  - secure: uvWMPRCu9Ww4aZOwjaECspxJ3NLQwLOhn1p5ND0mUADupOqFHmG6aiejhD+FoQ4JETd0FofHUAOhtIzmZuKYj08f/v7q6/bipSXqEj1Ufb0hgS7V9AaItfsYvgEeXr0jj04C1rfHCliSZGyDW2MhNoeNjIKgmyD9EbJ3lqpuHI4zXmWUrPknqqkq9GIN+ZI8AEGN6P1FtVD5zfQ43poWAiVlPVUiT6q/wcuDANx4teYPIBWC4jMNZRAP44JzMkru/TIlgM5V9UgvscBAHVt6V5U7fHolQJ+chwFioQniTySKkA5PyLnnXeydKyUajEzrylhQgBq9NuHtUs4cqEiop5RTwQ4yDQdCK9/1gY9FNRMvdSj0Z0C7YwA+m6YA95tAONf8PGAL1bXAQfXZ58IRMqyf/Gu+PTuOGVX/NPbXqYgWBXl9HAi4Es8pXl0EX3m3OSDbfaiXd+7ogxdiYiWiSjKUGfHsNpCii7bg2Txec3QnqzXPR1T+Vd3kNHX6z2Ry83K5z1UBwHoBkfSLOULY4mxI6zbhXU0h2fadLUDeCiTAe9KTYOWq4xpxiepbWjg2VBnN3Lw4inlevan2iw/kh9ZQzEurZUo/PakIHPWmUn2RbG7QWu88LNh8Ku80ei1lfVwk/8IpT9V4Clx9FRm4zedE1dDdsXy5OhKGJCk0llM=
```
Comment (sychan) [inline .travis.yml]: secure: "UPzlmxopnwD86fGPuyd7VNt1TNGxd9IwhCbtusd8PopsxYW064uJ5Ddy8AgkzE4gms5Bb+CIGgG5w/+TukkU0MTim/iZg6us6nKKcQXYUyGeTJvGx7NuSAlKaQjfZGj6kEM/xLMFnv51XKXCzU0rdNDGIVL5mEZ9x4HvcFYIArKofPsUuh52MbsdzTy9xQszEdkigBuAX1fU0sZmpClyNSVDbiZJQmlC//z8a0jbaZAa66vhDLYK1bBDtbUvdwc8eiu6x1qKJ4lOdZAjr+oRRARJFVTckhz8lpXXExTnipFOYvLRI8woxtNMI4ZyZ2ALkr1bPz7UqJ+URlKFu4+fTxLFik/7QIGZB4Nn67oBsVTbUAESrfOqu2ihfYsRTNdkxgb6H4ubWXLjJHznIDkddLV7uX53awfd9fJ8Vik0ZjHDUwOR4xMHkux5nFZZD6hrXkXu7eg8/o6eYNpdwXylZMnf81KVYLn8KegLppIY3L5bI0rYCXrhuHcHDypI4Ir2YqE63NxWTmAg0KQW+R0d6CzMipVfbH1lwSR/eyjdPktFhjGGvehZAKE7KP+jnHG7Tdf/HTb/Jgvlggz1KqQSuouOGoqylKUJbI9fVrEpwrD3qKLc76fsSXBmE2J0EHW9mSEwODwgx5S8P17I8xUMYAnS0jAE51ErBS7zJ6AZ3s0="

Comment (arfathpasha) [reply .travis.yml]: @sychan yes the encrypted "secure" tokens are kbase tokens. I realized from your comment that I should have also encrypted the codecov token as well. 
There is already a kbase account in codecov that we used for auth2. I am not sure who has access to this account. Here it is
https://codecov.io/gh/kbase/auth2

**Comment by sychan** [comment]
@arfathpasha I looked more carefully at things, sorry for responding before fully reviewing things.
I've requested updates to the travis.yml file that should enable tokens that work with the kbase/KBaseSearchEngine repo. If you can make those changes we can see if the travis runs go through.
We don't have a single email address for notificiations, since this repo is Gavin's baby, I would recommend that we either don't bother with email notifications and rely on the default Travis/Github notifications, or else put Gavin in there @MrCreosote do you have an opinion on this?
I've logged into Codecov with my account and gave Codecov access to the kbase organization repos. Here's the URL for KBase:
https://codecov.io/gh/kbase

Was there an additional level of integration that you were thinking of? It looks like CodeCov doesn't need anything special for public github repos.

File: .travis.yml (hunk 7 above)
Comment (sychan) [reply .travis.yml]: @arfathpasha The requested changes are for travis encrypted vars that should work for this repo. Can you make the changes on your PR?

**Comment by MrCreosote** [comment]
@sychan @arfathpasha 
> put Gavin in there @MrCreosote do you have an opinion on this?
Do whatever we did for auth2 I guess.

**Pull request merged**